Vagrant instance with basic Ansible

bombaywalla opened the pull request:
This PR includes a Vagrant instance using the ansible provisioner to bootstrap ansible onto the vagrant instance, followed by an ansible-local provisioner to use ansible (on the vagrant instance)to install some basic OS packages and Clojure.

bombaywalla committed:
Just after vagrant init ubuntu/bionic64'.

bombaywalla committed:
Added *.retry to gitignore since ansible.

bombaywalla committed:
Configure vagrant guest memory and set up provisioners.

bombaywalla committed:
Playbook to bootstrap ansible on remote host.

bombaywalla committed:
First cut at Ansible installing the basics.

bombaywalla committed:
Added python3.

plexus commented:
This looks great @bombaywalla, could you also add a section to the README explaining how to use the Vagrant setup?

bombaywalla committed:
Documented Vagrant instructions.

bombaywalla committed:
Install lein in /usr/local/bin rather than using apt.

bombaywalla committed:
Setup users and harden sshd.

bombaywalla committed:
Move the firewall settings into a role.

bombaywalla committed:
Added a note to see if some users should be removed.

lispyclouds commented on clojurians_log/playbooks/roles/clojure/tasks/main.yml:
```diff
@@ -0,0 +1,23 @@
+---
```
Do we need this role? As we are installing Leiningen already and would probably be building an uberjar from the source and running it with `java -jar ...`. Leiningen will get the Clojure jar automatically when building the project. What do you think @plexus ?

bombaywalla replied:
The clojure role installs the clojure CLI tools (the clj and clojure commands).
So, perhaps not needed. Unless we want to move to tools.deps rather than lein later.

lispyclouds replied:
I think we can remove it and use it when we actually need it. What do you think?

bombaywalla replied:
OK. The clojure role is removed.

bombaywalla commented:
Okay. Makes sense. Will remove. YAGNI! Thanks.

On Sun, May 5, 2019 at 12:08 PM Rahuλ Dé <notifications@github.com> wrote:

> *@lispyclouds* commented on this pull request.
> ------------------------------
>
> In clojurians_log/playbooks/roles/clojure/tasks/main.yml
> <https://github.com/clojureverse/nebula/pull/1#discussion_r281034554>:
>
> > @@ -0,0 +1,23 @@
> +---
>
> I think we can remove it and use it when we actually need it. What do you
> think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/clojureverse/nebula/pull/1#discussion_r281034554>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABF3IQ3JNBO7BC3BCHVLDKLPT4WBLANCNFSM4HJROT7A>
> .
>

bombaywalla committed:
Removed the installation of the Clojure CLI tools.

Based on discussion with @lispyclouds. May bring it back later if
needed. Currently, we use lein. Lein will download the necessary
Clojure jars. So, there is no need for installing the CLI tools at
this time.

bombaywalla committed:
Added htop & allowed http/s in firewall.

bombaywalla committed:
Fixed formatting for calling the firewall role.

I seem to have messed it up when I removed the call to the clojure
role.
